kore/src/Kore/Exec/GraphTraversal.hs

```diff
@@ -0,0 +1,338 @@
+{-# LANGUAGE MultiWayIf #-}
+
+{- |
+Copyright   : (c) Runtime Verification, 2018-2022
+License     : BSD-3-Clause
+Maintainer  : jost.berthold@runtimeverification.com
+-}
+module Kore.Exec.GraphTraversal (
+    graphTraversal,
+    simpleTransition,
+    Step,
+    TransitionResult (..),
+    TraversalResult (..),
+    transitionWithRule,
+) where
+
+import Control.Monad (foldM)
+import Control.Monad.Trans.State
+import Data.Limit
+import Data.List.NonEmpty qualified as NE
+import Data.Sequence (Seq (..))
+import Data.Sequence qualified as Seq
+import GHC.Natural
+import Kore.Rewrite.Strategy (
+    -- FIXME assimilate definitions and remove import
+    FinalNodeType (..),
+    GraphSearchOrder (..),
+    LimitExceeded (..),
+    unfoldSearchOrder,
+ )
+import Kore.Rewrite.Transition (
+    TransitionT (..),
+    runTransitionT,
+ )
+import Kore.Simplify.Data (Simplifier)
+import Prelude.Kore
+import Pretty
+
```

Comment:
I think this module would benefit from some extra documentation. My first thought was, how do `TransitionResult`, `StepResult` and `TraversalResult` all relate to eachother?

Reply:
I have added documentation that describes all these types together, in a haddock comment on `graphTraversal`. Is this what you had in mind?